File: python/Recursion/Factorial.py
# ******************************************
#  Author : Ali Azhari 
#  Created On : Tue Apr 20 2019
#  File : recursion.py
# *******************************************/

def fact(n):

    # Running time of factorial: O(n)
    if n < 0:
        raise Exception("Illegal argument Exception")
    if n == 0:
        return 1

    return n * fact(n - 1)

def Main():
    for i in range(11):
        print(f'Factorial of {i} :  {fact(i)}')

if __name__ == "__main__":
    Main()

File: Algorithms/java/Recursion/Factorial.java
package Recursion;

/**
 * @author Ali Azhari
 *
 */
public class Factorial {

	public static int Fact(int n) throws IllegalArgumentException {
		
		// Running time of factorial: O(n) 
		
		if (n < 0) throw new IllegalArgumentException();
		
		if (n == 0) return 1;
		
		return (n * Fact(n - 1));
		
	}
	
	public static void main(String[] args) {
		
		for (int i = 0; i < 11; i++)
		System.out.println("Factorial of " + i + " :" + Fact(i));

	}

}
